programs/mango-v4/src/instructions/perp_edit_market.rs

```diff
@@ -52,6 +49,48 @@ pub fn perp_edit_market(
     settle_pnl_limit_window_size_ts_opt: Option<u64>,
     reduce_only_opt: Option<bool>,
 ) -> Result<()> {
+    let group = ctx.accounts.group.load()?;
+
+    if oracle_opt.is_none()
+        && oracle_config_opt.is_none()
+        && base_decimals_opt.is_none()
+        && maint_base_asset_weight_opt.is_none()
+        && init_base_asset_weight_opt.is_none()
+        && maint_base_liab_weight_opt.is_none()
+        && init_base_liab_weight_opt.is_none()
+        && maint_pnl_asset_weight_opt.is_none()
+        && init_pnl_asset_weight_opt.is_none()
+        && liquidation_fee_opt.is_none()
+        && maker_fee_opt.is_none()
+        && taker_fee_opt.is_none()
+        && min_funding_opt.is_none()
+        && max_funding_opt.is_none()
+        && impact_quantity_opt.is_none()
+        && group_insurance_fund_opt.is_none()
+        && fee_penalty_opt.is_none()
+        && settle_fee_flat_opt.is_none()
+        && settle_fee_amount_threshold_opt.is_none()
+        && settle_fee_fraction_low_health_opt.is_none()
+        && stable_price_delay_interval_seconds_opt.is_none()
+        && stable_price_delay_growth_limit_opt.is_none()
+        && stable_price_growth_limit_opt.is_none()
+        && settle_pnl_limit_factor_opt.is_none()
+        && settle_pnl_limit_window_size_ts_opt.is_none()
+        // security admin can bring to reduce only mode
+        && reduce_only_opt.is_some()
```

Comment:
I'm concerned about code like this because someone will add a new field to perp_edit_market and forget to list it here.

Alternate suggestion:
```
let mut require_group_admin = false;

if let ... {
   ...
   require_group_admin = true; // do this on every single change that needs the group admin
}

// lots more seting of fields

// at the end
if require_group_admin {
   require!(...)
}
```